include/spdlog/details/pattern_formatter_impl.h

```diff
@@ -461,13 +466,21 @@ class full_formatter SPDLOG_FINAL : public flag_formatter
         msg.raw.str());*/
 
         // Faster (albeit uglier) way to format the line (5.6 million lines/sec under 10 threads)
-        msg.formatted << '[' << static_cast<unsigned int>(tm_time.tm_year + 1900) << '-'
-                      << fmt::pad(static_cast<unsigned int>(tm_time.tm_mon + 1), 2, '0') << '-'
-                      << fmt::pad(static_cast<unsigned int>(tm_time.tm_mday), 2, '0') << ' '
-                      << fmt::pad(static_cast<unsigned int>(tm_time.tm_hour), 2, '0') << ':'
-                      << fmt::pad(static_cast<unsigned int>(tm_time.tm_min), 2, '0') << ':'
-                      << fmt::pad(static_cast<unsigned int>(tm_time.tm_sec), 2, '0') << '.'
-                      << fmt::pad(static_cast<unsigned int>(millis), 3, '0') << "] ";
+        msg.formatted << '[';
```

Comment:
I split this line up because I was getting compilation issues I didn't understand.

Reply:
Probably makes sense to remove the performance note from the comment above, since the implementation has changed. Better yet, re-time the two versions with fmt 5.0.0, pick the fastest and update the comment.

Reply:
I don't think there is a way in ftm 5.0 to do it any way.
@vitaut Is there a way in fmt 5.0 achieve something similiar to (without using format strings):
```c++
fmt::memory_buffer buf;
buf << fmt::pad(static_cast<unsigned int>(tm_time.tm_min), 2, '0') << ':' <<..
```

Reply:
> Is there a way in fmt 5.0 achieve something similiar to (without using format strings)

No, the write API is gone for good so you'll have to use the format API or write to `memory_buffer` directly e.g. `buf.push_back(':')`.

Reply:
@vitaut I did find a workaround which is twice as fast (I checked).
Would be happy if you could quickly review is see if reasonable (espescailly regarding the  the use of the internal ```format_int``` class):
```c++
void append_int(int n, fmt::memory_buffer &dest)
{
    fmt::format_int i(n);
    dest.append(i.data(), i.data() + i.size());
}

inline void append_and_pad2(int n, fmt::memory_buffer &dest)
{
    if (n < 0) // should not happen in this formatter, but just in case, let fmt deal with negatives correctly
    {
        fmt::format_to(dest, "{:02}", n);
        return;
    }
    if (n < 10)
    {
        dest.push_back('0');
    }
    append_int(n, dest);
}

void append_and_pad3(int n, fmt::memory_buffer &dest)
{... }
...
...
        msg.formatted.push_back('[');
        append_int(tm_time.tm_year + 1900, msg.formatted);
        msg.formatted.push_back('-');

        append_and_pad2(tm_time.tm_mon + 1, msg.formatted);
        msg.formatted.push_back('-');

        append_and_pad2(tm_time.tm_mday, msg.formatted);
        msg.formatted.push_back(' ');

        append_and_pad2(tm_time.tm_hour, msg.formatted);
        msg.formatted.push_back(':');

        append_and_pad2(tm_time.tm_min, msg.formatted);
        msg.formatted.push_back(':');

        append_and_pad2(tm_time.tm_sec, msg.formatted);
        msg.formatted.push_back('.');

        append_and_pad3(static_cast<int>(millis), msg.formatted);
        msg.formatted.push_back(']');
        msg.formatted.push_back(' ');
...

Reply:
Looks fine to me. `format_int` is experimental (and undocumented), but still part of the public API and OK to use.